### DISCUSSION TITLE
How do I add or update a dropdown option value once it is created

### DISCUSSION THREAD

**Discussion by gzh1960** [Budibase/budibase#5565] [Help] [answered]
How do I add or update a dropdown option value once it is created?

Defined Budibase table and create a column with 4 option values.
I now need to add another value, how do I do this please?

Thanks in advance!

**Comment by melohagan** [comment, marked as answer]
Hi @gzh1960 

If you just want to add a static option to an existing Options Data Type, then you can simply click edit on the column:
![Screenshot 2022-04-26 at 10 23 10](https://user-images.githubusercontent.com/101575380/165268058-1133f5a3-74e9-45a3-96c5-3c5c13467b77.png)

If you want to dynamically add options, the easiest way to do this is to have a separate data table for your options. 
Each text type row will be an option.
![Screenshot 2022-04-26 at 11 11 24](https://user-images.githubusercontent.com/101575380/165277458-b9fdb6b7-a181-42d5-8a17-de66f1fe1a14.png)

Then you can use a data provider as source: https://docs.budibase.com/docs/option-picker#using-a-data-provider 
Adding options would be as easy as saving a row to this table.

----
**I strongly recommend the easy method above, but here's another way to do it FYI**

If for some reason you are forced to use the Options data type, then you can use the public API to update the table schema.

To add an option here, you would need to use the API to update the table schema: https://docs.budibase.com/reference/put_tables-tableid

Use the GET endpoint to grab the current schema, and then PUT the same schema, but with any number of additional options.

**Payload from the GET**
![Screenshot 2022-04-26 at 10 36 46](https://user-images.githubusercontent.com/101575380/165270637-9d65ac5a-8608-4f17-87ea-0ea816fb8781.png)

This would involve using  
- REST Queries: https://docs.budibase.com/docs/rest
- Execute Query action: https://docs.budibase.com/docs/actions

**Reply by gzh1960** [reply]
Thanks for the info. Yes had considered a table holding the values but for a few entries it seemed an overkill. Also I had entered data and then discovered I needed another option value.

The API option is another option but for non tech users could be an issue for them to use.

I think since it is so easy to add a few options, say less than 10,  when creating a table then you should be able to add or change the option values through the UI too. Please consider this if possible.

**Reply by melohagan** [reply]
You can change the options values when you edit the option column. Just enter a new row into the text box? Maybe I don't understand.

**Reply by gzh1960** [reply]
On table definition you can define the column option values, however once you save the column and add data to the table there does not seem to be any way to get that column under the Data menu to edit or add values. If I missing something here please advise.

**Reply by melohagan** [reply]
1. Go to the **Data** tab.
2. Click on your table
3. Hover the mouse over the Options column
4. Click the pencil icon
5. Change the option values and save (as you would when creating the row)

Does that work? If not, could you send a screenshot of where you are trying to edit.

**Reply by gzh1960** [reply]
My bad!! Thank you so much, yes, if I hover over the column and click the pencil icon next to it bam, I can edit the values.

Well done. Thanks again for your help!

**Reply by melohagan** [reply]
No worries! Glad to help :)
